main.tf

```diff
@@ -109,26 +123,17 @@ resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_id" {
 
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_crn" {
   count  = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
-  name   = "${var.prefix}-dns-binding"
+  name   = var.prefix != null ? "${var.prefix}-dns-binding" : var.dns_binding_name
```

Comment:
Regardless of the value of `var.prefix`, if the user passes a value for `var.dns_binding_name` the code should use that. Otherwise use "${var.prefix}-${var.name}-dns-binding" unless prefix is null, in which case use "${var.name}-dns-binding"